.github/workflows/dependencies-management.yml

```diff
@@ -76,12 +85,24 @@ jobs:
       # This action triggers the overall-tests.yml workflow on the PR
       # to allow the tests to run on the new dependencies.
       - name: Run tests on PR
-        uses: actions/github-script@v6
+        uses: actions/github-script@v7
         with:
+          github-token: ${{ secrets.TRIGGER_GITHUB_PR }}
           script: |
-            github.rest.actions.createWorkflowDispatch({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              workflow_id: 'overall-tests.yml',
-              ref: 'dependencies/update-python${{matrix.python-version}}',
-            })
+            const runTests = require('.github/scripts/run-workflow.js')
+            const linkTests = require('.github/scripts/link-workflow-to-pr.js')
```

Comment:
Two questions: 
1. Why in js and not in Python like other scripts?
2. Why not move this into a specific file?

Reply:
To be able to use the native "Github" (technically, it is [octokit](https://github.com/octokit/octokit.js/)) object, which simplifies GitHub interactions.

I created two files under the `.github` folder for clarity - reading/updating code in the middle of a YAML is not friendly - and save workflow size by 50 lines. Unfortunately, there is limitation and I need some code in the middle.

Reply:
oki